src/snowflake/snowpark/dataframe_ai_functions.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from snowflake.snowpark._internal.utils import (
+    create_prompt_column_from_template,
+    experimental,
+)
+from snowflake.snowpark.column import Column
+from snowflake.snowpark.functions import ai_complete as ai_complete_function
+from snowflake.snowpark._internal.telemetry import add_api_call
+
+if TYPE_CHECKING:
+    import snowflake.snowpark
+
+
+class DataFrameAIFunctions:
+    """Provides AI-powered functions for a :class:`DataFrame`."""
+
+    def __init__(self, dataframe: "snowflake.snowpark.DataFrame") -> None:
+        self._dataframe = dataframe
+
+    @experimental(version="1.37.0")
+    def complete(
+        self,
+        prompt: str,
+        input_columns: Union[List[Column], Dict[str, Column]],
+        output_column: Optional[str] = None,
```

Comment:
emmm, since this `output_column` is optional, do you want to be a keyword argument (put after *)?